src/threading.ts

```diff
@@ -61,11 +68,16 @@ class Thread{
 
     // Change Thread status
     public setStatus(status: enums.ThreadStatus): void {
+        // Fix for a single thread terminating multiple times in run(), whenever its stack is empty
```

Comment:
Hmm... I'll have to address this later. A terminated thread should never be resurrected, although I understand why that is happening now.